server/data/data-jmap/src/test/java/org/apache/james/jmap/api/change/MailboxChangeRepositoryContract.java

```diff
@@ -99,6 +99,17 @@ default void getChangesShouldReturnEmptyWhenNoNewerState() {
             .isEmpty();
     }
 
+    @Test
+    default void getChangesShouldReturnCurrentStateWhenNoNewerState() {
+        MailboxChangeRepository repository = mailboxChangeRepository();
+
+        MailboxChange oldState = MailboxChange.of(ACCOUNT_ID, STATE_0, DATE, ImmutableList.of(TestId.of(1)), ImmutableList.of(), ImmutableList.of());
+        repository.save(oldState);
+
+        assertThat(repository.getSinceState(ACCOUNT_ID, STATE_0, Optional.empty()).block().getNewState())
+            .isEqualTo(oldState.getState());
+    }
```

Comment:
I don't think so.

We need to differenciate the 
 - 'since state not found' => cannotCanCalculateChanges case
 -  from the 'no new changes' => return 0 changes


Mono.empty() vs Mono.just(empty()) should do the the trick.